src/ActionKit/CsrfTokenProvider.php

```diff
@@ -0,0 +1,65 @@
+<?php
+namespace ActionKit;
+use ActionKit\CsrfToken;
+use Exception;
+
+class CsrfTokenProvider {
+
+    public $tokenSessionId = '_csrf_token';
+    public $timeout = 300;
+
+    public function __construct($tokenSessionId = '_csrf_token', $timeout=300){
+        $this->tokenSessionId = $tokenSessionId;
+        $this->timeout = $timeout;
+    }
+
+    public function generateToken() {
+        $token = new CsrfToken($this->tokenSessionId, $this->timeout);
+        $token->time = time();
+        $token->salt = $this->randomString(32);
+        $token->sessid = session_id();
+        $token->ip = $_SERVER['REMOTE_ADDR'];
+
+        $_SESSION[$token->tokenSessionId] = serialize($token);
+
+        $hash = $this->calculateHash($token);
+        return base64_encode($hash);
```

Comment:
Make this method return the `Token` object and extract the code to another method:

```php
static public function encodeToken(CsrfToken $token) {
        $hash = self::calculateHash($token);
        return base64_encode($hash);
}
```